jenkins/image_building/initrd_sdk/unlock-mount-luks.sh

```diff
@@ -99,68 +159,304 @@ _dry_run() {
+fi
+
+# Config variables, only configurable via the config file:
+# key management config
+key_script="${key_script:-/etc/tpm2-unseal-key.sh}"
+auth="${auth:-secret}"
+secret_address="${secret_address:-0x81010002}"
+# root partition config
+root_dev_part_path="${root_dev_part_path:-$(_detect_root)}"
+# config drive partition config
+config_drive_dev_path="${config_drive_dev_path:-$(_detect_config)}"
+config_drive_part_num="${config_drive_part_num:-}"
+# dependency checks confg
+preparation_timeout="${preparation_timeout:-5}"
+# test config
+dry_run="${dry_run:-false}"
+no_preparation="${no_preparation:-false}"
+
+# key_command is always evaluated at the last suitable moment in order
+# to retain the file descriptor.
+key_command="${key_script} ${secret_address} ${auth} ${dry_run}"
 
 printf "INFO: unlock script has been started with the following arguments:\n"
-printf "Root device:%s, key script:%s, config-drive partition number:%s, " \
-    "${root_device_path}" "${key_script}" "${config_drive_part_num}"
-printf "dry run:%s\n" "${dry_run}"
+printf "root partition dev path:%s\nconfig-drive partition number:%s\n" \
+    "${root_dev_part_path:-auto}" "${config_drive_part_num:-auto}"
+printf "config-drive partition dev path:%s\nkey script:%s\nkey command:%s\n"\
+    "${config_drive_dev_path:-auto}" "${key_script}" "${key_command}"
+printf "DRY RUN MODE:%s\n" "${dry_run}"
+
+# If there is not root partition target there is no reason to continue
+if [[ -z "${root_dev_part_path}" ]]; then
+    printf "ERROR: no root partition was specified, auto detection failed!\n"
+    exit 1
+fi
+
+# Checking need for decryption and generating config drive metadata
+root_partition="$(_get_partition_from_blkid "${root_dev_part_path}")"
+part_prefix="$(_get_part_prefix_for_device "${root_partition}")"
+root_device="$(_find_disk_for_partition "${root_partition}")"
+part_count="$(_count_parts_for_disk "${root_device}")"
+config_drive_common="/dev/${root_device}${part_prefix}"
+if [[ -z "${config_drive_part_num}" ]] && [[ -z "${config_drive_dev_path}" ]];
+then
+    config_drive_dev_path="${config_drive_common}${part_count}"
+elif [[ -n "${config_drive_part_num}" ]] &&
+     [[ -z "${config_drive_dev_path}" ]];
+then
+    config_drive_dev_path="${config_drive_common}${config_drive_part_num}"
+fi
+printf "INFO: preview of generated configuration drive metadata:\n"
+printf "config-drive:%s\ndisk:%s\npart_count:%s\n" \
+    "${config_drive_dev_path}" "${root_device}" "${part_count}"
+printf "root_partition:%s\nprefix:%s\n" "${root_partition}" \
+    "${part_prefix}"
+if [[ "/dev/${root_partition}" == "${config_drive_dev_path}" ]]; then
+    printf "WARNING: Config drive candidate matches the root partition!\n"
+    printf "WARNING: Real config drive was not detected!\n"
+    config_drive_dev_path=""
+fi
+set +e
+_is_luks "${root_dev_part_path}"
+is_root_crypt=$?
+printf "INFO: root encrypted: %d\n" "${is_root_crypt}"
```

Comment:
this risks to be not very clear, maybe assign at least a yes or no, or change the message based on _is_luks result instead of just assigning the return value?

Reply:
I was contemplating about this, right now the function has no "regular" output just exit codes and this way it is also usable in "if" conditions . Originally these functions were used only in IF conditions but now they have been moved to be used like this.

But I agree with you this is not a nice syntax and might confuse folks, I will change this.

Reply:
Done, thanks for the suggestion!